llvm/lib/Transforms/Scalar/LICM.cpp

```diff
@@ -2803,15 +2803,19 @@ static bool hoistMulAddAssociation(Instruction &I, Loop &L,
 
 /// Reassociate associative binary expressions of the form
 ///
-/// 1. "(LV op C1) op C2" ==> "LV op (C1 op C2)"
+/// 1. "(LV op C1) op C2" ==> "LV op (C1 op C2)" if op is an associative BinOp
```

Comment:
I think it would have been sufficient here to copy 2. up, because the rest of the comment applies to both cases -- we don't need to separately explain the meaning of LV/C1/C2 for each case.